apps/settings/js/modules/base/event_emitter.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * EventEmitter implements basic event emitting logics. It also provides
+ * function that augments the ability to other objects. For example:
+ *  
+ *   function Module() {
+ *     this.EventEmitter.call(this, [eventName1, eventName2]);
```

Comment:
This line makes me confused. Why we have to predefined `eventName1` & `eventName2` before using ? Without this line, `this._eventListeners` will never be `{}` and would break other functionalities, so I can understand that you need this line to initialize it, but I think we should directly make `this._eventListeners = {}` when this `Object` is augmented.